### PR TITLE
Fix multiple monitor API bugs

### LINKED ISSUE #5316: Multiple monitor API errors
### Description

The monitor dashboard shows an error `TypeError: Cannot read properties of undefined (reading 'localeCompare')` with the latest monitor API. There are also a number of other related errors.

### Steps to reproduce

1. Run monitor API with testnet URL
2. Run monitor dashboard

### Additional context

_No response_

### Hedera network

testnet

### Version

0.73.0

### Operating system

None

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Fix race condition in environments with multiple clusters by retrying 404
* Fix race condition in environments with multiple clusters by retrying successful calls with empty lists
* Fix `TypeError: Cannot read properties of undefined (reading 'localeCompare')` in dashboard
* Fix `ReferenceError: blockUrl is not defined`
* Increase retry maxAttempts from 2 to 4
* Remove adhoc stateproof retry in favor of centralized 404 retry

**Related issue(s)**:

Fixes #5316

**Notes for reviewer**:

Will cherry pick to `release/0.74`.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/monitoring/monitor_apis/utils.js [hunk 1]
```diff
@@ -106,9 +116,10 @@ const fetchWithRetry = async (url, opts = {}) => {
  * Host info is prepended to if only path is provided
  * @param {*} url rest-api endpoint
  * @param {String} key JSON key of the object to return
+ * @param {Function} retryPredicate An optional predicate that returns true if a successful request should be retried.
  * @return {Object} JSON object representing api response or error
  */
-const getAPIResponse = async (url, key = undefined) => {
+const getAPIResponse = async (url, key = undefined, retryPredicate = undefined) => {
```
Comment (xin-hedera) [inline hedera-mirror-rest/monitoring/monitor_apis/utils.js:122]: can add a default noRetry predicate `const noRetry = () => false;`, then set as the default value here

```suggestion
const getAPIResponse = async (url, key = undefined, retryPredicate = noRetry) => {
```

then in `fetchWithRetry` can change to

```
      if (!retryPredicate(json)) {
        return json;
      }
```

File: hedera-mirror-rest/monitoring/monitor_apis/utils.js [hunk 2]
```diff
@@ -161,6 +167,14 @@ class ServerTestResult {
   }
 }
 
+/**
+ * Builds a retry predicate that checks if the JSON response contains a non-empty list.
+ *
+ * @param jsonRespKey The field within the JSON response that contains the list to check.
+ * @returns function A retry predicate
+ */
+const hasListItems = (jsonRespKey) => (res) => !res.hasOwnProperty(jsonRespKey) || res[jsonRespKey].length < 1;
```
Comment (xin-hedera) [inline hedera-mirror-rest/monitoring/monitor_apis/utils.js:176]: believe the function name should be either `hasEmptyList` or `hasNoListItems`

File: hedera-mirror-rest/monitoring/monitor_apis/account_tests.js [hunk 3]
```diff
@@ -235,18 +236,18 @@ const getSingleAccountTokenRelationships = async (server) => {
...
+    return {url: balances_url, ...balancesResult};
   }
   const accountId = balances[0].account;
 
   // Use that account id to call the /accounts/${accountId}/tokens endpoint
   const accountsTokenPath = `/accounts/${accountId}/tokens`;
   const accountsLimit = 1;
   let url = getUrl(server, accountsTokenPath, {limit: accountsLimit, order: 'asc'});
-  const tokenRelationships = await getAPIResponse(url, tokensJsonRespKey);
+  const tokenRelationships = await getAPIResponse(url, tokensJsonRespKey, hasListItems(jsonRespKey));
```
Comment (xin-hedera) [inline hedera-mirror-rest/monitoring/monitor_apis/account_tests.js:250]:
```suggestion
  const tokenRelationships = await getAPIResponse(url, tokensJsonRespKey, hasListItems(tokensJsonRespKey));
```

File: hedera-mirror-rest/monitoring/monitor_apis/token_tests.js [hunk 4]
```diff
@@ -384,7 +385,7 @@ const getTokenBalancesForAccount = async (server) => {
   }
 
   let url = getUrl(server, tokenBalancesPath(tokenId), {limit: 1});
-  let balances = await getAPIResponse(url, tokenBalancesJsonRespKey);
+  let balances = await getAPIResponse(url, tokenBalancesJsonRespKey, hasListItems(jsonRespKey));
```
Comment (xin-hedera) [inline hedera-mirror-rest/monitoring/monitor_apis/token_tests.js:388]:
```suggestion
  let balances = await getAPIResponse(url, tokenBalancesJsonRespKey, hasListItems(tokenBalancesJsonRespKey));
```

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/monitoring/monitor_apis/account_tests.js (hunk 3 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/monitoring/monitor_apis/account_tests.js]: Thanks. That's what I get for doing last minute refactoring.

File: hedera-mirror-rest/monitoring/monitor_apis/utils.js (hunk 2 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/monitoring/monitor_apis/utils.js]: Renamed to hasEmptyList

File: hedera-mirror-rest/monitoring/monitor_apis/utils.js (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/monitoring/monitor_apis/utils.js]: Done

File: hedera-mirror-rest/monitoring/monitor_apis/token_tests.js (hunk 4 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/monitoring/monitor_apis/token_tests.js]: Fixed

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by edwin-greene — APPROVED** [review]
Looks good

**Pull request merged**